### PR TITLE
Template update v2.0.1

### PR DESCRIPTION
**Pull request by ggabernet**
- Update to nf-core tools template v2.0.1

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/bcellmagic/tree/master/.github/CONTRIBUTING.md)
    - [x] If necessary, also make a PR on the nf-core/bcellmagic _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by ggabernet** [comment]
This is a huge PR again, to ease a bit the review, I'd like to say that modifications to `R` and `python` scripts that come shipped with the pipeline are strictly only to fix trailing spaces and indentation 4 spaces formatting, in accordance to the new `editorconfig` settings

**Comment by ggabernet** [comment]
The rest of the files have also only been strictly modified to adhere to the new template.

File: CITATIONS.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+* [SHazaM, Change-O](https://doi.org/10.1093/bioinformatics/btv359)
+    > Gupta, N. T., Vander Heiden, J. A., Uduman, M., Gadala-Maria, D., Yaari, G., & Kleinstein, S. H. (2015). Change-O: a toolkit for analyzing large-scale B cell immunoglobulin repertoire sequencing data: Table 1. Bioinformatics, 31(20), 3356–3358.
+
+* [Alakazam](https://doi.org/10.1126/scitranslmed.3008879)
+    > Stern, J. N. H., Yaari, G., Vander Heiden, J. A., Church, G., Donahue, W. F., Hintzen, R. Q., … O’Connor, K. C. (2014). B cells populating the multiple sclerosis brain mature in the draining cervical lymph nodes. Science Translational Medicine, 6(248).
+* [TIgGER](https://doi.org/10.1073/pnas.1417683112)
+    > Gadala-maria, D., Yaari, G., Uduman, M., & Kleinstein, S. H. (2015). Automated analysis of high-throughput B-cell sequencing data reveals a high frequency of novel immunoglobulin V gene segment alleles. Proceedings of the National Academy of Sciences, 112(8), 1–9.
+
```
Comment (FriederikeHanssen) [inline CITATIONS.md:28]: Tigger is below in the 'old' citation list in the README, but not here.

File: modules/local/get_software_versions.nf [hunk 2]
```diff
@@ -22,9 +24,9 @@ process GET_SOFTWARE_VERSIONS {
 
     input:
     path versions
-    
+
     output:
-    path "software_versions.csv"     , emit: csv
+    path "software_versions.tsv"     , emit: csv
```
Comment (FriederikeHanssen) [inline modules/local/get_software_versions.nf:29]: I guess this could be named tsv in the future at some point :D

File: nextflow.config [hunk 3]
```diff
@@ -102,130 +99,130 @@ includeConfig 'conf/modules.config'
...
-  description = 'B and T cell repertoire analysis pipeline with the Immcantation framework.'
-  mainScript = 'main.nf'
-  nextflowVersion = '>=21.04.1'
-  version = '2.0.0'
+    name            = 'nf-core/bcellmagic'
+    author          = 'Gisela Gabernet, Simon Heumos, Alexander Peltzer'
+    homePage        = 'https://github.com/nf-core/bcellmagic'
+    description     = 'B and T cell repertoire analysis pipeline with the Immcantation framework.'
+    mainScript      = 'main.nf'
+    nextflowVersion = '!>=21.04.1'
```
Comment (FriederikeHanssen) [inline nextflow.config:193]: Is this right? not greater/equal to the newest release?

**Review by FriederikeHanssen — APPROVED** [review]
Phew that is a huge template update 🙈 Looks awesome! Couple of minor comments from my side.

File: nextflow.config (hunk 3 above)
Comment (apeltzer) [reply nextflow.config:193]: 21.04.1 is the latest stable - 21.06 is an edge release I think ...

Comment (FriederikeHanssen) [reply nextflow.config:193]: Ah yes makes sense, I somehow thought about this differently. Lower threshold then is implictly the first stable nextflow release that supported DSL2...

File: modules/local/get_software_versions.nf (hunk 2 above)
Comment (ggabernet) [reply modules/local/get_software_versions.nf]: ok, just updated!

**Comment by ggabernet** [comment]
Thanks a lot for your review @FriederikeHanssen , just fixed all your points!

**Pull request merged**